src/utils/hooks/useAlerts/utils/utils.ts

```diff
@@ -0,0 +1,87 @@
+import { murmur3 } from 'murmurhash-js';
+
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import {
+  Alert,
+  AlertStates,
+  PrometheusRule,
+  Rule,
+  RuleStates,
+  Silence,
+  SilenceStates,
+} from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Group,
+  MONITORING_SALT,
+  PrometheusRulesResponse,
+} from '@virtualmachines/details/tabs/overview/utils/utils';
+
+export const addAlertIdToRule = (group: Group, rule: PrometheusRule): Rule => {
+  const key = [
+    group?.file,
+    group?.name,
+    rule?.name,
+    rule?.duration,
+    rule?.query,
+    ...Object?.entries(rule?.labels).map(([k, v]) => `${v}=${k}`),
+  ].join(',');
+  return { ...rule, id: String(murmur3(key, MONITORING_SALT)) };
+};
+
+export const getAlertsAndRules = (
+  data: PrometheusRulesResponse['data'],
```

Comment:
i think there is a data type for data

Reply:
I didn't see it anywhere. If you can find it I'll be happy to use it.

Reply:
can this help?
<img width="816" alt="image" src="https://user-images.githubusercontent.com/14824964/225600701-c43cca02-7e9e-4547-a05b-88f2d179f310.png">

Reply:
Unfortunately, no. This is what `PrometheusRulesResponse` looks like. The `data` field is different from `PrometheusResponse`. I don't know why they didn't create proper types for it on the console side.

```
export type PrometheusRulesResponse = {
  data: {
    groups?: Group[];
  };
  status: string;
};
```